code/backend/batch/utilities/helpers/DocumentProcessorHelper.py

```diff
@@ -55,12 +57,21 @@ def process(self, source_url: str, processors: List[Processor]):
                     documents: List[SourceDocument] = []
                     documents = document_loading.load(source_url, processor.loading)
                     documents = document_chunking.chunk(documents, processor.chunking)
-                    keys = list(map(lambda x: x.id, documents))
-                    documents = [
-                        document.convert_to_langchain_document()
-                        for document in documents
-                    ]
-                    return vector_store.add_documents(documents=documents, keys=keys)
+
+                    for document in documents:
+                        doc = {
+                            "id": document.id,
+                            "content": document.content,
+                            "content_vector": self.llm_helper.generate_embeddings(
+                                document.content
+                            ),
+                            # "metadata": document.metadata, this is just a full json escaped version of the entry
```

Comment:
Not really sure why this is needed. I can add this back later though.

Reply:
I'm happy with leaving a link to a new issue to investigate later if ok with you - does seem odd to have an escaped version to pass around.